src/main/java/io/confluent/idesidecar/restapi/models/ConnectionSpec.java

```diff
@@ -49,37 +221,271 @@ public String ccloudOrganizationId() {
 
   @Schema(description = "Configuration for Confluent Cloud connections")
   public record CCloudConfig(
-      @JsonProperty(value = "organization_id", required = true) String organizationId
+      @Schema(description = "The identifier of the CCloud organization to use. "
+                            + "The user's default organization is used when absent.")
+      @JsonProperty(value = "organization_id", required = true)
+      @NotEmpty
+      @Min(36)
+      @Max(36)
+      String organizationId
   ) {
-
   }
 
-  @Schema(description = "Configuration for local cluster")
+  @Schema(description = "Configuration when using Confluent Local and "
+                        + "optionally a local Schema Registry.")
   public record LocalConfig(
-      @JsonProperty(value = "schema-registry-uri") String schemaRegistryUri
+      @Schema(description = "The URL of the Schema Registry running locally.")
+      @JsonProperty(value = "schema-registry-uri")
+      @Null
+      @Max(512)
+      String schemaRegistryUri
+  ) {
+  }
+
+  @Schema(description = "Kafka cluster configuration.")
+  public record KafkaClusterConfig(
+      @Schema(description = "The identifier of the Kafka cluster, if known.")
+      @Null
+      @Max(64)
+      String id,
+
+      @Schema(description = "A list of host/port pairs to use for establishing the "
+                            + "initial connection to the Kafka cluster.")
+      @JsonProperty(value = "bootstrap_servers")
+      @NotEmpty
+      @Max(256)
+      String bootstrapServers
+  ) {
+
+    @JsonIgnore
+    public Optional<KafkaEndpoint> asCCloudEndpoint() {
```

Comment:
Thanks for adding in the support to "automatically" go from CCloud bootstrap servers -> Kafka REST URI, it's nice that we can construct it with some RegEx and string manipulation. 

User gives us CCloud bootstrap servers and a resource API key -- sidecan then has optionality on whether to talk to the cluster directly over native protocol or use the Kafka REST API. Sweet. 

Could you also shine some light on your rationale for adding this support?

Reply:
The primary reason is so that the user can create a direct connection to a CCloud Kafka or SR cluster, and we're able to infer the cluster's `id` value without connecting to it. And if the CCloud cluster is a Kafka cluster, then we can also infer the Kafka REST URL, for exposure in the GraphQL responses.

But then in the sidecar we can potentially optimize the CCloud scenarios by more easily getting the various ways to connect (e.g., CCloud APIs, Kafka REST URL, etc.), without necessarily having to extract information out of the plain `KafkaCluster` object, as that will already have been done.